xapian-core/cluster/cluster.cc

```diff
@@ -246,12 +236,15 @@ Point::initialize(TermListGroup &tlg, const Document &doc_) {
     for (TermIterator it = doc.termlist_begin(); it != doc.termlist_end(); ++it) {
 	doccount wdf = it.get_wdf();
 	string term = *it;
-	if (wdf < 1 || term[0]=='Z')
+	double termfreq = tlg.get_termfreq(term);
+	/** If term is not a stemmed term or the term indexes only one
+	 *  one document, do not compute TF-IDF scores for them
+	 */
+	if (wdf < 1 || term[0] != 'Z' || termfreq <= 1)
```

Comment:
FWIW:

```
vector<Point> points;
MSetPointSource ps(enq.get_mset(0, 10));

for (Xapian::PointIterator pit = ps.points_begin();
     pit != ps.points_end(); ++pit) {
    points.push_back(*pit);
}
```

could be probably be more succinctly written as:

```
MSetPointSource ps(enq.get_mset(0, 10));
vector<Point> points(ps.points_begin(), ps.points_end());
```

I think at this point we should finish off any remaining details and get what we have merged, rather than stepping back and potentially changing the design of the API.  We can certainly look at the bigger picture design in the light of having actually seen what the current approach looks like when filled out, but I don't think we want to delay merging for that - it'll just mean there's more pending stuff to have to worry about, and the risk of master and the branch diverging. (I discussed briefly with James on IRC that he's OK with this too).

> Using an `ExpandDecider` sounds like it would work, and I think would be a lot simpler than what I was proposing (the `wdf < 1` check would remain here, but that's to do with the compact vector calculation so it's fine).

The reason I wondered about another functor class was to allow it to consider term frequency and wdf too.  But I guess wdf == 0 is a special case (it typically means the term is boolean) and
so is a term frequency of 1 (because it means the term only occurs in one document, so we know it's useless for clustering).